_1327/documents/tests.py

```diff
@@ -473,6 +479,84 @@ def test_autosaves_removed_after_successful_edit(self):
+				reverse('documents:autosave', args=[url_title]),
+				params={'text': 'AUTO', 'title': form.get('title').value, 'comment': '', 'group': mommy.make(Group)},
+				xhr=True
+			)
+			self.assertEqual(response.status_code, 200)
+			self.assertEqual(TemporaryDocumentText.objects.count(), 1)
+			temporary_document_text = TemporaryDocumentText.objects.get()
+			self.assertIn('temp_', temporary_document_text.document.url_title)
+			temporary_document_text.delete()
+
+	def test_autosave_url_title_prefix_removed_on_save(self):
+		assign_perm("information_pages.add_informationdocument", self.group)
+		assign_perm("minutes.add_minutesdocument", self.group)
+		assign_perm("polls.add_poll", self.group)
+
+		for sub_class in Document.__subclasses__():
+			response = self.app.get(reverse('documents:create', args=[sub_class.__name__.lower()]), user=self.user)
+			self.assertEqual(response.status_code, 200)
+			form = response.forms['document-form']
+			# we need to scrape the url_title from the action as we can not get it from the url_title field in the form
+			url_title = re.search(r'/(?P<url_title>temp_.+)/', form.action).group(1)
+
+			# autosave AUTO
+			response = self.app.post(
+				reverse('documents:autosave', args=[url_title]),
+				params={'text': 'AUTO', 'title': form.get('title').value, 'comment': ''},
+				xhr=True
+			)
+			self.assertEqual(response.status_code, 200)
+
+			self.assertEqual(TemporaryDocumentText.objects.count(), 1)
+			temporary_document_text = TemporaryDocumentText.objects.get()
+			self.assertIn('temp_', temporary_document_text.document.url_title)
+
+			self.submit_document_form(form, sub_class)
+
+			self.assertEqual(TemporaryDocumentText.objects.count(), 0)
+			temp_prefix_len = re.search(r'temp_\d+_', url_title).end()
+			self.assertNotIn('temp_', Document.objects.get(url_title=url_title[temp_prefix_len:]).url_title)
+
+	def submit_document_form(self, form, sub_class):
+		text = 'Lorem Ipsum'
+		for field_name in form.fields.keys():
+			if field_name is not None and 'text' in field_name:
+				form.set(field_name, text)
+		form.set('comment', text)
+		form.set('group', self.group.pk)
+		if sub_class == MinutesDocument:
+			participants_field = form.get('participants')
+			participants_field.select_multiple([participants_field.options[0][0]])
+		response = form.submit().follow()
+		self.assertEqual(response.status_code, 200)
+
+	def test_autosave_url_title_correct_count_on_save(self):
+		assign_perm("information_pages.add_informationdocument", self.group)
+		assign_perm("minutes.add_minutesdocument", self.group)
+		assign_perm("polls.add_poll", self.group)
+
+		# create document with appended '_\d+' for testing a special case of url_title handling
+		mommy.make(InformationDocument, url_title=InformationDocument.__name__.lower())
+		test_document = mommy.make(InformationDocument, url_title="{}_2".format(InformationDocument.__name__.lower()))
+
+		self.assertEqual(
+			"{}_3".format(InformationDocument.__name__.lower()),
```

Comment:
i find this order confusing but it's not very important. i would prefer `assertEqual(needs_to_be_checked, 42)` because i read "assert that needs_to_be_checked is the expected value 42".